bindings/python/multibody/joint/joints-datas.hpp

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright (c) 2019 CNRS
+//
+
+#ifndef __pinocchio_python_joints_datas_hpp__
+#define __pinocchio_python_joints_datas_hpp__
+
+#include <eigenpy/exception.hpp>
+#include <eigenpy/eigenpy.hpp>
+
+namespace pinocchio
+{
+  namespace python
+  {
+    namespace bp = boost::python;
+
+
+    // generic expose_joint_data : do nothing special
+    template <class T>
```

Comment:
I would prefer to have struct specialization here, as the rest of the bindings.
Indeed they are more flexible and we allows to expose other Scalar types in the very near future.

Reply:
I don't understand what you mean. There is the struct exposer which calls the expose_joint_data. (and a similar one which calls the expose_joint_model. Would you prefer to move the expose_joint_* within the structs?

Reply:
You have done function specialization, but if we want to go towards genericity and being able to also expose code with automatic differentiation, we will need to use struct specialization which is less limited than function specialization.
I will take in charge this change latter inside another PR.

Reply:
Thanks